src/openvic-simulation/dataloader/Dataloader.cpp

```diff
@@ -46,8 +46,8 @@ constexpr bool path_equals(std::string_view lhs, std::string_view rhs) {
 }
 
 template<typename LT, typename RT>
-constexpr bool filename_equals(const LT& lhs, const RT& rhs) {
-	std::string_view left, right;
+bool filename_equals(const LT& lhs, const RT& rhs) {
+	std::string left, right;
```

Comment:
would this work instead:
```cpp
template<typename LT, typename RT>
bool filename_equals(const LT& lhs, const RT& rhs) {
	constexpr auto left = [&lhs] {
		if constexpr(std::same_as<LT, std::filesystem::path>)
			return lhs.filename().string();
		return lhs;
	};
	constexpr auto right = [&rhs] {
		if constexpr(std::same_as<RT, std::filesystem::path>)
			return rhs.filename().string();
		return rhs;
	};
	return path_equals(left(), right());
}
```
if this could work it would be preferable as it minimizes the need for construction

Reply:
Been trying to get this to work with `g++ 12.3` but it really doesn't like it, I'm gonna leave it for now and let you improve it in a future PR.